core/issuers.js

```diff
@@ -66,10 +85,67 @@ Description must be truthy: ${description}`;
       return mint.mint(assay.empty(), name); // mint and issuer call each other
     },
 
-    getExclusive(amount, srcPaymentP, name) {
+    combine(paymentsArray, name = 'combined payment') {
+      const totalAmount = paymentsArray.reduce((soFar, payment) => {
+        return assay.with(soFar, paymentController.getAmount(payment));
+      }, assay.empty());
+
+      const combinedPayment = makePayment(name);
+
+      // ///////////////// commit point //////////////////
+      // All queries above passed with no side effects.
+      // During side effects below, any early exits should be made into
+      // fatal turn aborts.
+      for (const payment of paymentsArray) {
+        paymentController.remove(payment);
+      }
+      paymentController.recordNew(combinedPayment, totalAmount);
+      return combinedPayment;
+    },
+
+    split(payment, amountsArray, namesArray) {
+      namesArray =
+        namesArray !== undefined
+          ? namesArray
+          : Array(amountsArray.length).fill('a split payment');
+
```

Comment:
We need a check that the namesArray (at this point) is the same length as the amountsArray.

Reply:
added